app/policies/configuration/zone_policy.rb

```diff
@@ -1,25 +1,27 @@
 class Configuration::ZonePolicy
-  def initialize(context, agent)
+  def initialize(context, zone)
     @current_agent = context.agent
-    @current_territory = context.territory
-    @agent = agent
+    @zone = zone
   end
 
   def territorial_admin?
-    @current_agent.territorial_admin_in?(@current_territory)
+    @current_agent.territorial_admin_in?(@zone.territory)
   end
 
   alias new? territorial_admin?
   alias create? territorial_admin?
   alias destroy? territorial_admin?
 
   class Scope
-    def initialize(context, _scope)
-      @current_territory = context.territory
+    def initialize(context, scope)
+      @current_agent = context.agent
+      @scope = scope
     end
 
     def resolve
-      @current_territory.zones
+      @scope
+        .joins(:sector)
+        .where(sectors: { territory_id: @current_agent.territorial_roles.pluck(:territory_id) })
```

Comment:
vu qu'on enlève le filtre sur le current territory dans la policy, je pense qu'il faut qu'on le rajoute dans le controller pour pas avoir de mauvaise surprise en cas d'agent multi-territoire

Reply:
bien vu 🙇

Reply:
done in https://github.com/betagouv/rdv-service-public/pull/4513/commits/de71f5ee0444dce446e919f6564082e486731bb0 , testé manuellement en local